syntax = "proto3";

package helloworld.v1;

// The greeting service definition.
service GreeterService {
  // Sends a greeting
  rpc SayHello(SayHelloRequest) returns (SayHelloResponse) {}

  rpc SayHelloStreamReply(SayHelloStreamReplyRequest) returns (stream SayHelloStreamReplyResponse) {}

  rpc SayHelloBidiStream(stream SayHelloBidiStreamRequest) returns (stream SayHelloBidiStreamResponse) {}
}

message SayHelloRequest {
  string name = 1;
}

message SayHelloResponse {
  string message = 1;
}

message SayHelloStreamReplyRequest {
  string name = 1;
}

message SayHelloStreamReplyResponse {
  string message = 1;
}

message SayHelloBidiStreamRequest {
  string name = 1;
}

message SayHelloBidiStreamResponse {
  string message = 1;
}
